### PR TITLE
test: add fuzzing test to LSP16

### PR DESCRIPTION
**Pull request by skimaharvey**
# What does this PR introduce? 

Fuzzing tests for LSP16: 

- [x] LSP16 should not keep ethers 
- [x] `calculateAddress` should calculate the right address 
- [x] `calculateProxyAddress` should calculate the right address  
- [x] `_generateSalt` should be unique  

While checking for the above properties, we are also checking the LSP16 works properly and that contracts are being deployed

### CODE REVIEW

File: contracts/Mocks/NonPayableFallback.sol [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+contract NonPayableFallback {
+    fallback() external {}
```
Comment (YamenMerhi) [inline contracts/Mocks/NonPayableFallback.sol:5]: Solhint warning

File: tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    }
+
+    function testCalculateAddressShouldReturnCorrectUPAddressWithDeployCreate2Init(
+        bytes32 providedSalt,
+        uint256 valueForInitializer
+    ) public {
+        vm.deal(address(this), valueForInitializer);
+        assert(address(this).balance == valueForInitializer);
+
+        bytes memory initializeCallData = hex"aabbcc";
```
Comment (YamenMerhi) [inline tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol:254]: Can't we pass this as a param and put `if ( bytes4(initializeCallData) != bytes4(0) || initializeCallData.length > 4 ) return` ? something like this. If not, its already good, its just a slight improvement

**Review by YamenMerhi — APPROVED** [review]

File: tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol (hunk 2 above)
Comment (skimaharvey) [reply tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol:254]: The thing is that I am afraid we will "loose" a lot of tests if the seeds decides to send lots of initializeCallData of lengthes > 4. 
What I probably do is something like `abi.encodePacked(bytes4(initializeCallData))`

File: tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+    uint256 public testCounter;
+
+    function setUp() public {
+        lsp16 = new LSP16Mock();
+
+        nonPayableFallbackContract = new NonPayableFallback();
+        fallbackInitializer = new FallbackInitializer();
+        lsp0Init = new LSP0ERC725AccountInit();
+        lsp0 = new LSP0ERC725Account(address(20));
```
Comment (CJ42) [inline tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol:38]: what does the `address(20)` do here? Curious as never saw this before

**Review by CJ42 — APPROVED** [review]

File: tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol (hunk 3 above)
Comment (skimaharvey) [reply tests/foundry/LSP16UniversalFactory/LSP16UniversalProfile.t.sol:38]: I dont know why I picked 20. Just used a random address as owner of LSP0ERC725Account 😅

**Pull request merged**